### PR TITLE
AK+LibCrypto: Make constructing a BigInteger from string fallible

### PR DESCRIPTION
**Pull request by tcl3**
Previously, constructing a `UnsignedBigInteger::from_base()` could produce an incorrect result if the input string contained a valid Base36 digit that was out of range of the given base. The same method would also crash if the input string contained an invalid Base36 digit. An error is now returned in both these cases.

Constructing a BigFraction from string is now also fallible, so that we can handle the case where we are given an input string with invalid digits.

The motivation for this change was to fix the behavior of Calculator when pasting invalid numbers. 

Video showing the issues:

https://github.com/SerenityOS/serenity/assets/2817754/e15fc9b4-1673-41eb-ab40-427f1695eedb

NB: I have used `MUST` for the usages of these functions in LibJS and LibWeb, as I believe they are all pre-validated, so shouldn't fail.

### CODE REVIEW

File: Userland/Libraries/LibWeb/HTML/StructuredSerialize.cpp [hunk 1]
```diff
@@ -936,7 +936,8 @@ class Deserializer {
         auto string_view = TRY(Bindings::throw_dom_exception_if_needed(vm, [&string]() {
             return string->utf8_string_view();
         }));
-        return JS::BigInt::create(vm, ::Crypto::SignedBigInteger::from_base(10, string_view.substring_view(0, string_view.length() - 1)));
+        auto bigint = MUST(::Crypto::SignedBigInteger::from_base(10, string_view.substring_view(0, string_view.length() - 1)));
```
Comment (LucasChollet) [inline Userland/Libraries/LibWeb/HTML/StructuredSerialize.cpp:939]: I have very little knowledge on what the code does, but the word deserialize make me think that this function can take user-provided data (the string check above also kinda confirm that). So we should probably use `TRY` here (probably with that `Bindings::throw_dom_exception_if_needed` that I see for the first time).

**Review by LucasChollet — COMMENTED** [review]
Nice to see someone come back for that FIXME
Code looks good, except one comment (keep in mind that `LibWeb` is far from my expertise)

File: Userland/Libraries/LibWeb/HTML/StructuredSerialize.cpp (hunk 1 above)
Comment (tcl3) [reply Userland/Libraries/LibWeb/HTML/StructuredSerialize.cpp:939]: Looking at #20132 made me think this is unlikely to fail in the normal course of events. I _think_ the correct way to propagate this kind of error would be something like:

```suggestion
        auto bigint_or_error = ::Crypto::SignedBigInteger::from_base(10, string_view.substring_view(0, string_view.length() - 1));
        if (bigint_or_error.is_error())
            return WebIDL::DataCloneError::create(*vm.current_realm(), "Invalid BigInt"_fly_string);
```

I'm not sure if that's the correct thing to do though? I also have very little experience with LibWeb.

Comment (ADKaster) [reply Userland/Libraries/LibWeb/HTML/StructuredSerialize.cpp:939]: This shouldn't fail due to invalid formatting of the data, only due to memory allocation failure.

The way that structured data is input into the system is only via the structured_serialize APIs. I don't know of a way currently beyond hacker craziness with a write primitive to get user-controlled data into the stream (unless we have bugs in serialziation/deserialization, e.g. they are not 100% symmetrical).

Comment (LucasChollet) [reply Userland/Libraries/LibWeb/HTML/StructuredSerialize.cpp:939]: It's ready for a merge then!

**Review by LucasChollet — APPROVED** [review]

**Pull request merged**